pandas/tests/groupby/test_apply.py

```diff
@@ -544,9 +544,10 @@ def filt2(x):
         else:
             return x[x.category == "c"]
 
-    expected = data.groupby("id_field").apply(filt1)
-    result = data.groupby("id_field").apply(filt2)
-    tm.assert_frame_equal(result, expected)
+    with pytest.raises(ValueError, match="buffer source array is read-only"):
```

Comment:
While a tradeoff since not sure how much it would complicate the implementation, would it be worth reraising an error specifically mentioning groupby mutation? Would worry a user would see this error in a new version and be confused since they know their data was not read-only going into the op.

Reply:
i think thatd be worthwhile yah